<launch>
	<arg name="joy_dev" default="/dev/input/js0" />
	<!-- Choose from: ps4_wired, ps4_wireless, elecom_small -->
	<arg name="joy_config" default="ps3"/>
	<arg name="joypad_type" default="ps4_wired"/>
	<arg name="config_filepath" default="$(find teleop_twist_joy)/config/$(arg joy_config).config.yaml" />
	<arg name="joy_topic" default="joy" />

	<node pkg="joy" type="joy_node" name="joy">
		<remap from="joy" to="$(arg joy_topic)" />
		<param name="dev" value="$(arg joy_dev)" />
		<param name="deadzone" value="0" />
		<param name="autorepeat_rate" value="100" />
	</node>
	<node pkg="teleop_twist_joy" name="teleop_twist_joy" type="teleop_node">
		<remap from="joy" to="$(arg joy_topic)" />
		<rosparam command="load" file="$(arg config_filepath)" />
	</node>
	<node pkg="tracked_mpc" name="tracked_mpc" type="dummy_bot_node" output="screen">
		<param name="parent_frame" value="world" />
		<param name="child_frame" value="master" />
	</node>
	<node pkg="rviz" name="rviz" type="rviz" args="-d $(find tracked_mpc)/launch/demo.rviz" output="screen"/>
</launch>